packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -1554,7 +1554,7 @@ class MaterialBasedCupertinoThemeData extends CupertinoThemeData {
       // like barBackgroundColor can still behave like a normal
       // CupertinoThemeData.
       super.raw(
-        _cupertinoOverrideTheme.brightness,
+        _cupertinoOverrideTheme.brightness ?? _materialTheme.brightness,
```

Comment:
This looks reasonable, but why aren't the other properties here defaulting to what is in the `_materialTheme` as well? Some may not have a reasonable default, but it seems weird to only do this for brightness.

Reply:
We should document this behavior somewhere (probably in the doc string for the `MaterialBasedCupertinoThemeData` class.

Reply:
I think `MaterialBasedCupertinoThemeData` wants to differentiate the part of the theme that comes from `_cupertinoOverrideTheme` from the rest. From `MaterialBasedCupertinoThemeData.copyWith`'s API doc:

>   /// Copies the [ThemeData]'s `cupertinoOverrideTheme`.
  ///
  /// Only the specified override attributes of the [ThemeData]'s
  /// `cupertinoOverrideTheme` and the newly specified parameters are in the
  /// returned [CupertinoThemeData]. No derived attributes from iOS defaults or
  /// from cascaded Material theme attributes are copied.
  ///
  /// [MaterialBasedCupertinoThemeData.copyWith] cannot change the base
  /// Material [ThemeData]. To change the base Material [ThemeData], create a
  /// new Material [Theme] and use `copyWith` on the Material [ThemeData]
  /// instead.